staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -363,6 +367,19 @@ func (r *Request) Timeout(d time.Duration) *Request {
 	return r
 }
 
+func (r *Request) RequestTimeout(d time.Duration) *Request {
```

Comment:
If I set a timeout and one isn't set, set it.

If I set a timeout and a shorter one is set, don't do anything.

If I set a timeout and a longer one is set, reset it to the shorter one.

Reply:
> If I set a timeout and a longer one is set, reset it to the shorter one.

Why would you reset it to shorter one, if someone explicitly asks for longer - use it.